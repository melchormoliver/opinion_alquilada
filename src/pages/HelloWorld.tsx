import React from 'react';
import {
  IonCol,
  IonContent,
  IonGrid,
  IonHeader,
  IonPage,
  IonRow,
  IonTitle,
  IonToolbar,
} from '@ionic/react';

import './HelloWorld.css';

const HelloWorld: React.FC = () => {
  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonTitle>Hello World2</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent>
        <IonGrid>
          <IonRow>
            <IonCol>
              <IonTitle size='large'>Hello World2</IonTitle>
            </IonCol>
          </IonRow>
        </IonGrid>
      </IonContent>
    </IonPage>
  );
};

export default HelloWorld;
