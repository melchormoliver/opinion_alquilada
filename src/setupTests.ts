// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import { mockIonicReact } from '@ionic/react-test-utils';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';

configure({ adapter: new Adapter() });
mockIonicReact();

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };
