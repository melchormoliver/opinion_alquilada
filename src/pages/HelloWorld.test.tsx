import '../setupTests';
import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import HelloWorld from './HelloWorld';

test('HelloWorld - renders without crashing', async () => {
  const { findByText } = render(<HelloWorld />);
  await findByText('Hello World2');
});

test('should test Header component', () => {
  const wrapper = shallow(<HelloWorld />);
  expect(wrapper).toMatchSnapshot();
});
