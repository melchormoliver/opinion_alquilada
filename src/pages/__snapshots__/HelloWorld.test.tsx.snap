// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Header component 1`] = `
<ForwardRef(IonPage)>
  <ForwardRef(IonHeader)>
    <ForwardRef(IonToolbar)>
      <ForwardRef(IonTitle)>
        Hello World2
      </ForwardRef(IonTitle)>
    </ForwardRef(IonToolbar)>
  </ForwardRef(IonHeader)>
  <ForwardRef(IonContent)>
    <ForwardRef(IonGrid)>
      <ForwardRef(IonRow)>
        <ForwardRef(IonCol)>
          <ForwardRef(IonTitle)
            className="colorRojo"
            size="large"
          >
            Hello World46465465465
          </ForwardRef(IonTitle)>
        </ForwardRef(IonCol)>
      </ForwardRef(IonRow)>
    </ForwardRef(IonGrid)>
  </ForwardRef(IonContent)>
</ForwardRef(IonPage)>
`;
